packages/atomic/cypress/integration/search-box/instant-results/search-box-instant-results.cypress.ts

```diff
@@ -59,6 +63,66 @@ describe('Instant Results Test Suites', () => {
       .init();
   }
 
+  describe('with a custom aria label', () => {
+    const customFieldName = 'custom';
+    const customFieldValueForResult = (index: number) => `${index}!`;
+    const ariaLabelGeneratorAlias = 'ariaLabelGenerator';
+    beforeEach(() => {
+      new TestFixture()
+        .with(setInstantResults(numOfInstantResults))
+        .with(setRecentQueries(numOfRecentQueries))
+        .withCustomResponse((response) => {
+          response.results.forEach(
+            (result, i) =>
+              (result.raw[customFieldName] = customFieldValueForResult(i))
+          );
+        })
+        .with(
+          addSearchBox({
+            suggestions: {maxWithoutQuery: 8, maxWithQuery: 8},
+            instantResults: {
+              ariaLabelGenerator: cy
+                .stub()
+                .as(ariaLabelGeneratorAlias)
+                .callsFake(
+                  <AriaLabelGenerator>(
+                    ((_, result) => result.raw[customFieldName] as string)
+                  )
+                ),
+            },
+          })
+        )
+        .init();
+      SearchBoxSelectors.inputBox().type(`${downKeys(2)}`, delay());
+    });
+
+    it('uses the generated labels', () => {
+      InstantResultsSelectors.results().should(([...results]) =>
+        results.forEach((result, i) =>
+          expect(result.ariaLabel).to.contain(customFieldValueForResult(i))
+        )
+      );
+    });
+
+    it('passes the bindings to the generator', () => {
+      cy.get<
+        sinon.SinonStub<
+          Parameters<AriaLabelGenerator>,
+          ReturnType<AriaLabelGenerator>
+        >
+      >(`@${ariaLabelGeneratorAlias}`).then((stub) =>
+        cy
+          .get('atomic-search-interface')
+          .then(([searchInterface]) => initializeBindings(searchInterface))
```

Comment:
For some reason, swapping the first an second get..then pairs caused both `then` to yield the value of the first `get`... wasted lots of time.